functional-test/src/test/groovy/org/rundeck/util/common/execution/ExecutionUtils.groovy

```diff
@@ -0,0 +1,89 @@
+package org.rundeck.util.common.execution
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.rundeck.util.api.responses.execution.Execution
+import org.rundeck.util.api.responses.jobs.JobExecutionsResponse
+import org.rundeck.util.container.RdClient
+
+class ExecutionUtils {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper()
+    static final List<String> EXECUTION_FINISH_STATUSES = [
+            'aborted',
+            'failed',
+            'succeeded',
+            'timedout',
+            'other'].asImmutable()
+
+    static class Verifiers {
+        /**
+         * Returns a closure that verifies if an execution passed into it is finished.
+         * @return
+         */
+        static final Closure<Boolean> executionFinished() {
+            // The verifier is untyped to be compatible with pre-existing code that uses Maps instead of typed objects
+            return {
+                EXECUTION_FINISH_STATUSES.contains(it?.status)
+            }
+        }
+
+
+        /**
+         * Returns a closure that verifies if an execution passed into it is running.
+         * @return
+         */
+        static final Closure<Boolean> executionRunning() {
+            // The verifier is untyped to be compatible with pre-existing code that uses Maps instead of typed objects
+            return {
+                ["running"].contains(it?.status)
+            }
+        }
+
+    }
+
+    static class Retrievers {
+        /**
+         * Returns a closure that retrieves an execution by its id.
+         * @param client
+         * @param executionId
+         * @return
+         */
+        static final Closure<Execution> executionById(RdClient client, String executionId) {
```

Comment:
This is not a request for change but just a thought: I'm realizing that these methods would end up being very handy outside of the waiting scope. Maybe we could explore a way for these to return the plain object instead of a Closure, while still being usable by the waiting api. Or maybe i'm dreaming too high :)

Reply:
Hmm... I think `waitForExecution` in the JobUtils does that. It returns an execution object and relies on this closure for behaviour implementation.